contracts/Reserve/AmmReserve/AmmReserve.cpp

```diff
@@ -224,29 +225,17 @@ void AmmReserve::trade(name from, asset src, string memo, name code, state &stat
 
     /* get conversion rate again */
     asset dest = asset();
-    double conversion_rate = reserve_get_conv_rate(src, buy, dest);
+    double charged_fee = 0;
+    double conversion_rate = reserve_get_conv_rate(src, buy, dest, charged_fee);
     eosio_assert(conversion_rate > 0, "conversion rate must be bigger than 0");
     eosio_assert(conversion_rate < MAX_RATE, "fail overflow validation");
 
-    record_profit(buy ? dest : src, buy);
     async_pay(_self, receiver, dest, dest_contract, "");
-}
 
-void AmmReserve::record_profit(asset token, bool buy) {
-    params_type params_inst(_self, _self.value);
-    auto params = params_inst.get();
-
-    double token_damount = amount_to_damount(token.amount, token.symbol.precision());
-    double dprofit = buy ?
-        (token_damount * params.profit_percent / (100.0 - params.profit_percent)) :
-        (token_damount * params.profit_percent) / 100.0;
-    int64_t profit_amount = damount_to_amount(dprofit, token.symbol.precision());
-    asset profit = asset(profit_amount, token.symbol);
-
-    state_type state_inst(_self, _self.value);
-    auto s = state_inst.get();
-    s.collected_profit_in_tokens += profit;
-    state_inst.set(s, _self);
+    asset charged_fee_asset = asset(damount_to_amount(charged_fee, EOS_PRECISION), EOS_SYMBOL);
```

Comment:
small optimization will be to check if charged_fee > 0 before doing all the conversions.